### DISCUSSION TITLE
PnP Core SDK File Upload Need help with Blazor WASM

### DISCUSSION THREAD

**Discussion by nikomas78** [pnp/pnpcore#410] [Q&A] [answered]
Is there anyone with an example showing how to pass the values into the file upload object using Blazor WASM?
I am unfamiliar with file upload and using OpenRead and how to pass the filepath and filecontent into the PnP object?

// Get a reference to a folder
IFolder siteAssetsFolder = await context.Web.Folders.Where(f => f.Name == "SiteAssets").FirstOrDefaultAsync();

// Upload a file by adding it to the folder's files collection
IFile addedFile = await siteAssetsFolder.Files.AddAsync("test.docx", 
                  System.IO.File.OpenRead($"{Path.DirectorySeparatorChar}TestFilesFolder{Path.DirectorySeparatorChar}test.docx"));

**Comment by jansenbe** [comment, marked as answer]
@nikomas78 : maybe this helps: https://blog.stevensanderson.com/2019/09/13/blazor-inputfile/

`file.Data` is a stream that can be used in the `Files.AddAsync` method as explained here: https://pnp.github.io/pnpcore/using-the-sdk/files-intro.html#adding-files-uploading

**Reply by nikomas78** [reply]
@jansenbe Thank you for explaining. I do have one other question about the appsettings.json. With the PnP Core Authentication in a Blazor WASM app, the appsettings.json file is exposed to the users. All the demos show this being the case. Is it standard for an Azure AD registered app's ClientId and TenantId be open? Not sure how to encrypt the config file values on the frontend?

Any suggestions or is this normal?

{
    "key": "generic",
    "SharePoint": {
      "SiteUrl": "https://mytenant.sharepoint.com"
    },
     "AzureAd": {
      "Authority": "https://login.microsoftonline.com/{TenantId}",
      "ClientId": "{ClientId}", 
      "ValidateAuthority": true
    },
    "GraphScopes": "User.Read;MailboxSettings.Read;Calendars.ReadWrite;User.ReadWrite;Sites.ReadWrite.All",
    "Logging": {
      "LogLevel": {
        "Default": "Information"
      }
    }
  }

**Reply by jansenbe** [reply]
@nikomas78 : typically that's fine to share those an that's what other apps that are client side only do as well. One can get the tenant id of any tenant fairly easily, and getting the app id any application that triggers auth from a client device is simple as well.
